esphome/components/wifi/__init__.py

```diff
@@ -483,3 +486,36 @@ async def wifi_enable_to_code(config, action_id, template_arg, args):
 @automation.register_action("wifi.disable", WiFiDisableAction, cv.Schema({}))
 async def wifi_disable_to_code(config, action_id, template_arg, args):
     return cg.new_Pvariable(action_id, template_arg)
+
+
+@automation.register_action(
+    "wifi.set_sta",
```

Comment:
Not sure I like the action name being `set_sta`. This refers to an internal function name and is not exactly user friendly for the yaml side. 
Maybe simply `wifi.connect`?
```suggestion
    "wifi.connect",
```

Reply:
I feel like `connect` could be confused with just enabling/disabling it. Maybe `update`, `change`, or even just `set`?

Reply:
This is true, maybe `wifi.configure`?